### PR TITLE
{docs} reference guidelines - removed markdown syntax not supported in this repo

### PR DESCRIPTION
**Pull request by dbradish-microsoft**
- reformatted checklist UI as [!div class="checklist"] is not supported in this repo
- fixed site-relative links not supported in this repo
- Substituted contractions where possible
- Slight changes to `Minimum content` H2 syntax

### CODE REVIEW

**Comment by yonzhan** [comment]
docs

**Comment by dbradish-microsoft** [comment]
#sign-off

File: doc/reference_doc_guidelines.md [hunk 1]
```diff
@@ -188,23 +186,19 @@ examples:
                --ssh-key-values @key-file
   ```
 
-- Not all examples can be shell agnostic due to the quoting differences between Bash and PowerShell. For examples that only run in Bash, provide a second example for PowerShell. If you are testing examples in PowerShell, here are some considerations:
-  - [Quoting issues with PowerShell](https://github.com/Azure/azure-cli/blob/dev/doc/quoting-issues-with-powershell.md)
```
Comment (jsntcy) [inline doc/reference_doc_guidelines.md:192]: Did you remove them because they are already at the beginning of this file?

**Comment by jsntcy** [comment]
Overall LGTM.

File: doc/reference_doc_guidelines.md (hunk 1 above)
Comment (dbradish-microsoft) [reply doc/reference_doc_guidelines.md:192]: @jsntcy, yes

**Review by jsntcy — APPROVED** [review]

**Pull request merged**